In [1]:
import pandas as pd

In [2]:
df_1=pd.merge(pd.read_csv("FF_Processed_Data_a.csv"),(pd.read_csv("FF_Processed_Data_b.csv")),on='idnum')
print('pair 1...')
df_2=pd.merge((pd.read_csv("FF_Processed_Data_c.csv")),(pd.read_csv("FF_Processed_Data_d.csv")),on='idnum')
print('pair 2...')
df_3=pd.merge(df_1,df_2,on='idnum')
print('almost...')
df=pd.merge(df_3,(pd.read_csv("FF_Processed_Data_e.csv")),on='idnum')
df

pair 1...
pair 2...
almost...


Unnamed: 0_x_x  idnum  p6a1_int  p6a2_1_int  p6a2_2_int  p6a2_3_int  \
0                 0      1       NaN         NaN         NaN         NaN   
1                 1      2       NaN         NaN         NaN         NaN   
2                 2      3       NaN         NaN         NaN         NaN   
3                 3      4       NaN         NaN         NaN         NaN   
4                 5      6       NaN         NaN         NaN         NaN   
5                 7      8       NaN         NaN         NaN         NaN   
6                 8      9       NaN         NaN         NaN         NaN   
7                 9     10       NaN         NaN         NaN         NaN   
8                10     11       NaN         NaN         NaN         NaN   
9                11     12       NaN         NaN         NaN         NaN   
10               12     13       NaN         NaN         NaN         NaN   
11               13     14       NaN         NaN         NaN         NaN   
12               14     15       NaN         NaN         NaN         NaN   
13               15     16       NaN         NaN         NaN         NaN   
14               17     18       NaN         NaN         NaN         NaN   
15               18     19       NaN         NaN         NaN         NaN   
16               20     21       NaN         NaN         NaN         NaN   
17               21     22       4.0         NaN         NaN         NaN   
18               22     23       NaN         NaN         NaN         NaN   
19               23     24       NaN         NaN         NaN         NaN   
20               24     25       NaN         NaN         NaN         NaN   
21               26     27       NaN         NaN         NaN         NaN   
22               27     28       NaN         NaN         NaN         NaN   
23               28     29       NaN         NaN         NaN         NaN   
24               29     30       NaN         NaN         NaN         NaN   
25               30     31       NaN         NaN         NaN         NaN   
26               32     33       NaN         NaN         NaN         NaN   
27               35     36       NaN         NaN         NaN         NaN   
28               38     39       NaN         NaN         NaN         NaN   
29               39     40       NaN         NaN         NaN         NaN   
..              ...    ...       ...         ...         ...         ...   
297             402    403       NaN         NaN         NaN         NaN   
298             403    404       NaN         NaN         NaN         NaN   
299             404    405       NaN         NaN         NaN         NaN   
300             405    406       NaN         NaN         NaN         NaN   
301             406    407       NaN         NaN         NaN         NaN   
302             408    409       NaN         NaN         NaN         NaN   
303             409    410       NaN         NaN         NaN         NaN   
304             410    411       NaN         NaN         NaN         NaN   
305             413    414       NaN         NaN         NaN         NaN   
306             414    415       NaN         NaN         NaN         NaN   
307             415    416       NaN         NaN         NaN         NaN   
308             416    417       NaN         NaN         NaN         NaN   
309             417    418       NaN         NaN         NaN         NaN   
310             418    419       NaN         NaN         NaN         NaN   
311             419    420       NaN         NaN         NaN         NaN   
312             423    424       NaN         NaN         NaN         NaN   
313             424    425       NaN         NaN         NaN         NaN   
314             426    427       NaN         NaN         NaN         NaN   
315             429    430       NaN         NaN         NaN         NaN   
316             430    431       NaN         NaN         NaN         NaN   
317             432    433       NaN         NaN     

In [11]:
reps=list(df.keys())
for x in reps:
    for y in reps:
        if abs(df[x].corr(df[y], min_periods=100))>0.8 and x!=y and x[2]!='n' and x[3]!='i'and y[3]!='i'and y[2]!='n':
            print(x+' and '+y+' are corr by '+str(df[x].corr(df[y])))
print('Done!')

p6f1_int and k6c23_int are corr by 0.8162812101257854
p6f6_int and p6f8_int are corr by 0.8119900321696658
p6f8_int and p6f6_int are corr by 0.8119900321696658
p6g6_int and p6k57_int are corr by 0.8436332687773416
p6j34_int and p6j35_int are corr by 0.8913418598218212
p6j35_int and p6j34_int are corr by 0.8913418598218212
p6k20_int and p6k31_int are corr by 0.9127988943467346
p6k20_int and p6k57_int are corr by 0.8505791766836605
p6k31_int and p6k20_int are corr by 0.9127988943467346
p6k31_int and p6k57_int are corr by 0.8693869009207299
p6k57_int and p6g6_int are corr by 0.8436332687773416
p6k57_int and p6k20_int are corr by 0.8505791766836603
p6k57_int and p6k31_int are corr by 0.8693869009207299
ck6pcgrel_int and ck6mcrel_int are corr by 0.9166242444241173
ck6pcgrel_int and ck6conf1d_int are corr by 1.0
ck6yagey_int and ck6yagem_int are corr by 0.9398072797740419
ck6yagey_int and ch6yagey_int are corr by 0.8922411773875178
ck6yagey_int and ch6yagem_int are corr by 0.8941040680049815

### Overall Strongest Corrs

- p6e1_int and ck6conf2_int are corr by 0.7763937476407247
- p6g6_int and p6k57_int are corr by 0.8436332687773416
- k6b32e_int and k6f13_int are corr by 0.9180571207748202
- k6c23_int and p6f1_int are corr by 0.8162812101257854
- k6f13_int and k6b32e_int are corr by 0.9180571207748202
- k6f83_int and o6a6e_int are corr by 1.0
- o6a7c_int and o6c1_int are corr by 0.8125233379769967
- o6d1_1_int and o6f4_int are corr by -0.8013709421212104
- o6g3_int and o6g2_int are corr by 0.8104908904947038
- k6d62b_int and k6d62c_int are corr by 0.6715231543091531
- k6f13_int and k6b1a_int are corr by 0.6016742987052753
- k6f13_int and k6b8_int are corr by 0.6928241978507362
- p6e8_int and p6e27_1_int are corr by -0.6319875961975814
- budget stuff with ps and k/j/g etc.
- o6e9_int and o6e11_int are corr by 0.7038255255177734
- o6c1_int and k6b8_int are corr by -0.6827467057135748
- k6c1_int and ck6conf2_int are corr by 0.6517515625618875
- k6f12_int and ck6chtcm_int are corr by 0.6392627491874527
- k6f18a_int and p6b17_int are corr by 0.70541943573372 (example of weird corrs)
- p6h102_int and o6c5_int
- p6f1_int and k6c23_int are corr by 0.8162812101257854